test/widgets/search_ride_page_test.dart

```diff
@@ -0,0 +1,967 @@
+          await pumpMaterial(tester, const SearchRidePage());
+
+          await tester.tap(find.byKey(const Key('searchRideFilterButton')));
+          await tester.pump();
+
+          final Finder featuresExpandButton = find.byKey(const Key('searchRideFeaturesExpandButton'));
+          await tester.tap(featuresExpandButton);
+          await tester.pump();
+
+          final Finder featureChip = find.byKey(Key('searchRideFeatureChip${Feature.values[0].name}'));
+          await tester.tap(featureChip);
+          await tester.pump();
+
+          await tester.tap(featuresExpandButton);
+          await tester.pump();
+
+          expectFeaturesVisible(tester, [Feature.values[0]]);
+
+          await tester.tap(featureChip);
+          await tester.pump();
+
+          expectFeaturesVisible(tester, [Feature.values[0]]);
+        });
+
+        testWidgets('Mutually exclusive', (WidgetTester tester) async {
+          final List<Feature> shuffledFeatures = [...Feature.values]..shuffle();
+          final Feature mutuallyExclusive1 = shuffledFeatures.firstWhere((Feature feature) =>
+              Feature.values.firstWhereOrNull((Feature other) => feature.isMutuallyExclusive(other)) != null);
+          final Feature mutuallyExclusive2 =
+              shuffledFeatures.firstWhere((Feature feature) => feature.isMutuallyExclusive(mutuallyExclusive1));
+
+          await pumpMaterial(tester, const SearchRidePage());
+
+          await tester.tap(find.byKey(const Key('searchRideFilterButton')));
+          await tester.pump();
+
+          await tester.tap(find.byKey(const Key('searchRideFeaturesExpandButton')));
+          await tester.pump();
+
+          await tester.tap(find.byKey(Key('searchRideFeatureChip${mutuallyExclusive1.name}')));
+          await tester.pump();
+          await tester.tap(find.byKey(Key('searchRideFeatureChip${mutuallyExclusive2.name}')));
+          await tester.pump();
+
+          expect(find.byKey(const Key('searchRideFeatureMutuallyExclusiveSnackBar')), findsOneWidget);
+        });
+      });
+
+      testWidgets('Filter rating and features', (WidgetTester tester) async {
+        final DateTime startTime = DateTime.now();
+
+        const int minRating = 3;
+        final List<Feature> shuffledFeatures = [...Feature.values]..shuffle();
+        final List<Feature> mutuallyExclusiveFiltered = [];
+        for (final Feature feature in shuffledFeatures) {
+          if (mutuallyExclusiveFiltered.firstWhereOrNull((Feature other) => feature.isMutuallyExclusive(other)) ==
+              null) {
+            mutuallyExclusiveFiltered.add(feature);
+          }
+        }
+        final List<Feature> requiredFeatures =
+            mutuallyExclusiveFiltered.sublist(random.integer(mutuallyExclusiveFiltered.length - 1));
+
+        final List<Profile> drivers = [
+          //Satisfies everything
+          ProfileFactory().generateFake(
```

Comment:
Just define these with names that say what they are doing, then you can remove the comments. (and the code with `drivers[3]` is more readable, for example)